financial-connections/src/main/java/com/stripe/android/financialconnections/browser/CustomTabsManager.kt

```diff
@@ -0,0 +1,163 @@
+package com.stripe.android.financialconnections.browser
+
+import android.app.Activity
+import android.content.ComponentName
+import android.content.Context
+import android.content.Intent
+import android.net.Uri
+import androidx.browser.customtabs.CustomTabsCallback
+import androidx.browser.customtabs.CustomTabsClient
+import androidx.browser.customtabs.CustomTabsIntent
+import androidx.browser.customtabs.CustomTabsServiceConnection
+import androidx.browser.customtabs.CustomTabsSession
+import com.stripe.android.core.Logger
+
+/**
+ * Manages the connection to the CustomTabsService.
+ *
+ */
+internal class CustomTabsManager(
+    private val logger: Logger
+) {
+
+    private var client: CustomTabsClient? = null
+    private var connection: CustomTabsServiceConnection? = null
+    private var session: CustomTabsSession? = null
+
+    /**
+     * Binds the Activity to the CustomTabsService.
+     */
+    fun onStart(context: Context) {
+        if (client != null) return
+        connection = object : CustomTabsServiceConnection() {
+            override fun onCustomTabsServiceConnected(
+                name: ComponentName,
+                client: CustomTabsClient
+            ) {
+                log("Service connected")
+                this@CustomTabsManager.client = client
+                this@CustomTabsManager.client?.warmup(0)
+                session = this@CustomTabsManager.client?.newSession(CustomTabsCallback())
+            }
+
+            override fun onServiceDisconnected(name: ComponentName) {
+                log("Service disconnected")
+                client = null
+                session = null
+            }
+        }
+        bindCustomTabService(context)
+    }
+
+    /**
+     * Unbinds the Activity from the CustomTabsService.
+     */
+    fun onStop(context: Context) {
```

Comment:
You could consider using a `LifecycleObserver` to close the connection automatically.

Reply:
https://github.com/stripe/stripe-android/pull/7081/commits/d239616ac4746b3c6d3d3247c8177d53045f7e77